swarm/src/lib.rs

```diff
@@ -457,6 +459,24 @@ where
         SwarmBuilder::without_executor(transport, behaviour, local_peer_id).build()
     }
 
+    /// Adds a layer on the `Swarm` that logs all trafic that passes through the sockets
+    /// created by it.
+    ///
+    /// This method returns an `Arc<BandwidthSinks>` that can be used to retreive the total number
+    /// of bytes transferred through the sockets.
+    pub fn with_bandwidth_logging(mut self) -> Arc<BandwidthSinks> {
+        let sinks = BandwidthSinks::new();
+        let sinks_copy = sinks.clone();
+        self.transport = Transport::map(self.transport, |(peer_id, stream_muxer_box), _| {
+            (
+                peer_id,
+                StreamMuxerBox::new(BandwidthLogging::new(stream_muxer_box, sinks_copy)),
+            )
+        })
+        .boxed();
+        sinks
+    }
```

Comment:
I don't think this should live at the `Swarm` level. Instead, I think we can have this in `libp2p` same as before but implement `TransportExt` for every `Transport` that outputs `(PeerId, S)` with `S: StreamMuxer`. Does that compile?

Reply:
Yep, thanks!